src/cloud-api-adaptor/install/yamls/caa-pod.yaml

```diff
@@ -1,91 +0,0 @@
-apiVersion: apps/v1
```

Comment:
@stevenhorsman,

By removing this file won't we break the TLS and CRI runtime endpoint overlays?

If you enable any of them in the `kustomization.yaml`. it will try to find a DaemonSet with "peerpodconfig-ctrl-caa-daemon" name, but is missing within the current execution, even tho it will be managed by

```
$ kubectl apply -k src/cloud-api-adaptor/src/cloud-api-adaptor/install/overlays/libvirt/
# Warning: 'patchesStrategicMerge' is deprecated. Please use 'patches' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
error: no matches for Id DaemonSet.v1.apps/peerpodconfig-ctrl-caa-daemon.confidential-containers-system; failed to find unique target for patch DaemonSet.v1.apps/peerpodconfig-ctrl-caa-daemon.confidential-containers-system
 ```
 
 Did I missed something?

Reply:
I'll take a look today. Recently I've been using the scripts I updated rather than manually, so I probably missed this error, though the fact everything still seems to work is interesting.

Reply:
I can't reproduce this and doing all the steps manually still don't see the error:
```
# kubectl apply -k "github.com/confidential-containers/operator/config/overlays/peerpods/default"
# kubectl set env -n confidential-containers-system deployment/cc-operator-controller-manager RELATED_IMAGE_CAA=${CAA_IMAGE}
# kubectl rollout status deploy/cc-operator-controller-manager -n confidential-containers-system
# kubectl apply -k "github.com/confidential-containers/operator/config/samples/ccruntime/peer-pods"
# kubectl apply -k install/overlays/libvirt/
serviceaccount/cloud-api-adaptor created
clusterrole.rbac.authorization.k8s.io/node-viewer created
clusterrole.rbac.authorization.k8s.io/peerpod-editor created
clusterrole.rbac.authorization.k8s.io/pod-viewer created
clusterrolebinding.rbac.authorization.k8s.io/node-viewer created
clusterrolebinding.rbac.authorization.k8s.io/peerpod-editor created
clusterrolebinding.rbac.authorization.k8s.io/pod-viewer created
configmap/peer-pods-cm created
secret/auth-json-secret created
secret/peer-pods-secret created
secret/ssh-key-secret created
```

Let's chat when you are online and see where the differences are

Reply:
@stevenhorsman you need to enable the patch here:

https://github.com/confidential-containers/cloud-api-adaptor/blob/main/src/cloud-api-adaptor/install/overlays/libvirt/kustomization.yaml#L60

Reply:
Beraldo has let me know that we need to uncomment the cri-runtime patch in https://github.com/confidential-containers/cloud-api-adaptor/blob/0db87d03d6887884d2c27a0b86dccc1a51bfa8f7/src/cloud-api-adaptor/install/overlays/libvirt/kustomization.yaml#L60 and with that I can now reproduce the error:
```
# kubectl apply -k install/overlays/libvirt/
# Warning: 'patchesStrategicMerge' is deprecated. Please use 'patches' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
error: no matches for Id DaemonSet.v1.apps/peerpodconfig-ctrl-caa-daemon.confidential-containers-system; failed to find unique target for patch DaemonSet.v1.apps/peerpodconfig-ctrl-caa-daemon.confidential-containers-system
```
I'm not 100% what it is needed for, but I guess we should try and resolve it if possible